### ISSUE TITLE
Create the Vanilla\AddonModel to enable/disable addons

### ISSUE THREAD

**Issue by tburry** [vanilla/vanilla#5108]
Depends on #5131.

**This class has largely been implemented in #5366, but it isn't a complete implementation to satisfy this issue.**

The Gdn_ApplicationManager and Gdn_PluginManager clasess acted like models as well as runtime managers. I don't want to add configuration management to the AddonManager, but rather add an Vanilla\AddonModel to correspond to the /settings/addons page that will manage addons within an application.

- This model must reside in the Vanilla namespace because the addons application already has an AddonModel.
- The Vanilla\AddonModel doesn't inherit Gdn_Model and will only have a few methods.
- The class should declare dependencies in its constructor. I think it needs to depend on AddonManager and Gdn_Configuration, but there may be more.
- Method getWhere($where) will have possible filters of **key**, **type** (for addon type) and **enabled**. Look at Addon::makeFilterCallback for help here. This method will return an array representation of the addons rather than Addon objects. Add an enabled key to the result. I want the new addons page to do mostly client-side filtering so this method doesn't have to be exhaustive.
- Method setEnabled($addonKey, $enabled) will enable/disable an addon and an addon's dependencies. It should return an array of addons that were affected with addonKey => enabled. For future work this will be better done, but first getting the array of addons affected and then enabling them. When disabling an addon that has other addons that depend on it an exception should be thrown. This method can use the Schema and ValidationException.
- The Vanilla\AddonModel will save to the EnabledAddons config key, but it should have bc to the other keys and migrate where appropriately.

**Comment by tburry** [comment]
I'm closing this issue since it's largely been completed now.
